components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/helpers/runtime/DefineZoneHelper.java

```diff
@@ -45,14 +47,20 @@ public CharSequence apply(String zoneName, Options options) throws IOException {
         buffer.append("<!--[UUF-ZONE]{\"name\": \"").append(zoneName).append("\",\"position\": \"start\"}-->\n");
 
         List<Fragment> bindings = lookup.getBindings(requestLookup.tracker().getCurrentComponentName(), zoneName);
-        if (!bindings.isEmpty()) {
-            API api = options.data(HbsRenderable.DATA_KEY_API);
-            for (Fragment fragment : bindings) {
-                buffer.append(fragment.render(new ContextModel(options.context), lookup, requestLookup, api));
+        Optional<String> zoneContent = requestLookup.getZoneContent(zoneName);
+        if (bindings.isEmpty() && !zoneContent.isPresent() && options.tagType == TagType.SECTION) {
+            // {{#defineZone "zone-name"}}default content{{/defineZone}}
+            buffer.append(options.fn().toString());
+        } else {
+            if (!bindings.isEmpty()) {
```

Comment:
`zoneContent.ifPresent(buffer::append);` in line 61 should be executed either `bindings` is empty or not. So IMO this cannot be merged into an else-if.